overlord/snapstate/snapstate.go

```diff
@@ -719,6 +719,25 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 	if err != nil {
 		return nil, err
 	}
+
+	// see if we need to update the channel
+	if infoErr != nil && snapst.Channel != channel {
+		snapsup := &SnapSetup{
+			SideInfo: snapst.CurrentSideInfo(),
+			Channel:  channel,
+		}
+		snapsup.SideInfo.Channel = channel
```

Comment:
I don't think we do this usually,  that value reflect what we got from the store, maybe I'm misremembering though